PhysicsTools/Heppy/python/analyzers/objects/JetAnalyzer.py

```diff
@@ -424,6 +425,31 @@ def isFlavour(x,f):
                    jet.mcFlavour = 0
 
         self.heaviestQCDFlavour = 5 if len(self.bqObjects) else (4 if len(self.cqObjects) else 1);
+
+
+    def makeNIsr(self,event):
+
+        event.nIsr = 0
+
+        for jet in self.cleanJetsAll:
+
+            if jet.pt()<30.0: continue
+            if abs(jet.eta())>2.4: continue
+            matched = False
+            for mc in event.genParticles:
+                if matched: break
+                if (mc.status()!=23 or abs(mc.pdgId())>5): continue
+                momid = abs(mc.mother().pdgId())
+                if not (momid==6 or momid==23 or momid==24 or momid==25 or momid>1e6): continue
```

Comment:
Not sure whether you know but this will e.g. not catch jets from DY events away from the Z peak where no Z boson appears in the event record. I don't know whether this is a practical issue for your analysis, but I'd suggest to use something more generic if you want to put this into common code.